env/etc/php/8.1/Dockerfile

```diff
@@ -0,0 +1,171 @@
+RUN apt-get update \
+    && apt-get install -y \
+    apt-utils \
+    wget \
+    libwebp-dev \
+    libxpm-dev \
+    libfreetype6-dev \
+    libjpeg62-turbo-dev \
+    libmcrypt-dev \
+    libpng-dev \
+    libxslt-dev \
+    libicu-dev \
+    mariadb-client \
+    pv \
+    vim \
+    nano \ 
+    bash-completion \
+    openssh-server \
+    ssl-cert \
+    msmtp \
+    sudo \
+    dnsutils \
+    iputils-ping \
+    iputils-tracepath \
+    host \
+    strace \
+    telnet \
+    unzip \
+    gnupg \
+    gcc \
+    lsof \
+    libsodium-dev \
+    libzip-dev \
+    && apt-get update \
+    && apt-get clean all
+
+#CREATE USER
+ENV _USER=magento
+ENV _HOME_DIRECTORY=/home/${_USER}
+RUN useradd -m ${_USER} && echo "${_USER}:${_USER}" | chpasswd && chsh ${_USER} -s /bin/bash && adduser ${_USER} sudo
+
+#PHP EXTENSIONS
+RUN docker-php-ext-install -j$(nproc) soap sockets \
+    && docker-php-ext-configure gd --enable-gd --with-freetype --with-jpeg --with-webp \
+    && docker-php-ext-install -j$(nproc) gd bcmath pdo_mysql xsl intl zip
+
+#GIT
+RUN apt-get update \
+    && apt-get install -y git \
+    && apt-get clean all
+
+#NODEJS
+RUN curl -sL https://deb.nodesource.com/setup_12.x | bash \
+    && apt-get install -y nodejs
+
+#BLACKFIRE
+RUN version=$(php -r "echo PHP_MAJOR_VERSION.PHP_MINOR_VERSION;") \
+    && curl -A "Docker" -o /tmp/blackfire-probe.tar.gz -D - -L -s https://blackfire.io/api/v1/releases/probe/php/linux/amd64/$version \
+    && mkdir -p /tmp/blackfire \
+    && tar zxpf /tmp/blackfire-probe.tar.gz -C /tmp/blackfire \
+    && mv /tmp/blackfire/blackfire-*.so $(php -r "echo ini_get('extension_dir');")/blackfire.so \
+    && printf "extension=blackfire.so\nblackfire.agent_socket=tcp://blackfire:8707\n" > $PHP_INI_DIR/conf.d/blackfire.ini \
+    && rm -rf /tmp/blackfire /tmp/blackfire-probe.tar.gz \
+    && mkdir -p /tmp/blackfire \
+    && curl -A "Docker" -L https://blackfire.io/api/v1/releases/client/linux_static/amd64 | tar zxp -C /tmp/blackfire \
+    && mv /tmp/blackfire/blackfire /usr/bin/blackfire \
+    && rm -Rf /tmp/blackfire
+
+#COMPOSER
+RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer1 --1\
+    && apt -qy install $PHPIZE_DEPS && mkdir /${_HOME_DIRECTORY}/.composer \
+
+#COMPOSER 2 (additional)
+RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer2 --2
+
+#COMPOSER link (2 is default for PHP 8.1)
+RUN ln -s /usr/local/bin/composer2 /usr/local/bin/composer
+
+#XDEBUG
+RUN pecl install xdebug \
+    && echo ";zend_extension=xdebug.so" >> /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini \
+    && echo "xdebug.remote_host=host.docker.internal" >> /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini \
+    && echo "xdebug.remote_enable=1" >> /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini \
+    && echo "xdebug.remote_autostart=1" >> /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini \
```

Comment:
remote_host, remote_enable and remote_autostart are not correct for xDebug3
instead of them, it should be:
xdebug.client_host=host.docker.internal
xdebug.mode=debug
xdebug.start_with_request=yes